AdyenComponents/BACS Direct Debit/Trackers/BACSDirectDebitComponentTracker.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import Foundation
+
+/// :nodoc:
+internal protocol BACSDirectDebitComponentTrackerProtocol: AnyObject {
+    func sendEvent()
+}
+
+/// :nodoc:
+internal class BACSDirectDebitComponentTracker: BACSDirectDebitComponentTrackerProtocol {
```

Comment:
this is over engineering, adds no value and complicates the code for no reason, its much simpler and straightforward to just call this line:

```
Analytics.sendEvent(component: paymentMethod.type,
                             flavor: isDropIn ? .dropin : .components,
                             context: apiContext)
```
in the implementation of the `TrackableComponent` implementation.